contracts/base/LenderInfo.sol

```diff
@@ -0,0 +1,216 @@
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        return true;
+    }
+
+    function withdrawInterest(address recipient, uint256 amount)
+        external
+        isDaiPool(msg.sender)
+        isValid(recipient)
+        returns (uint256)
+    {
+        // Updating accrued interest for zDAI recipient.
+        updateAccruedInterestFor(recipient);
+
+        uint256 amountToWithdraw = amount;
+        uint256 accruedInterest = lenderAccounts[recipient].totalAccruedInterest;
+        
+        if (accruedInterest >= amountToWithdraw) {
+            lenderAccounts[recipient].totalAccruedInterest = accruedInterest.sub(amountToWithdraw);
+        } else {
+            amountToWithdraw = lenderAccounts[recipient].totalAccruedInterest;
+            lenderAccounts[recipient].totalAccruedInterest = 0;
+        }
+        lenderAccounts[recipient].lastBlockAccrued = this.getCurrentBlockNumber();
+
+        emit AccruedInterestWithdrawn(recipient, amountToWithdraw);
+
+        return amountToWithdraw;
+    }
+
+    function getCurrentBlockNumber()
+        external
+        view
+        returns (uint256 blockNumber)
+    {
+        return block.number;
+    }
+
+    /** Internal Functions */
+
+    /**
+        Updates the accrued interest for a specific lender.
+    
+        @param lender address.
+        @return the updated accrued intereset for the lender address.
+     */
+    function updateAccruedInterestFor(address lender)
+        internal
+        returns (uint256)
+    {
+        // Get last block accrued
+        uint256 previousBlockAccruedInterest = lenderAccounts[lender].lastBlockAccrued;
+        uint256 currentBlockNumber = this.getCurrentBlockNumber();
+
+        // Update last block accrued.
+        lenderAccounts[lender].lastBlockAccrued = currentBlockNumber;
+
+        // Update total accrued interest.
+        lenderAccounts[lender].totalAccruedInterest = calculateNewAccruedInterestFor(
+                                                            lenderAccounts[lender].totalAccruedInterest,
+                                                            previousBlockAccruedInterest,
+                                                            currentBlockNumber,
+                                                            getZDaiBalanceOf(lender)
+                                                        );
+
+        emit AccruedInterestUpdated(
+            lender,
+            lenderAccounts[lender].lastBlockAccrued,
+            lenderAccounts[lender].totalAccruedInterest
+        );
+
+        return lenderAccounts[lender].totalAccruedInterest;
+    }
+
+    function calculateNewAccruedInterestFor(
+        uint256 currentAccruedInterest,
+        uint256 previousBlockAccruedInterest,
+        uint256 currentBlockNumber,
+        uint256 currentZDaiBalance
+    )
+        internal
+        pure
+        returns (uint256 newAccruedInterest)
```

Comment:
Naming the return parameter declares a variable called that. In this case you don't use the return parameter name, so you should remove the name. It's fine to name the return parameter if you are then using that variable within the function.